File: Examples/PackageDiagramDiff/Swift/After/Banking/InMemoryAccountRepository.swift
/// An in-memory `AccountRepository`. Part of the `Banking` module; depends on `Core`.
final class InMemoryAccountRepository: AccountRepository {
    private var storage: [String: Account] = [:]

    func find(id: String) -> Account? {
        storage[id]
    }

    func save(_ account: Account) {
        storage[account.id] = account
    }
}

File: Examples/PackageDiagramDiff/TypeScript/After/Banking/InMemoryAccountRepository.ts
import { Account } from "../Core/Account";
import { AccountRepository } from "../Core/AccountRepository";

/** An in-memory `AccountRepository`. Part of the `Banking` module; depends on `Core`. */
export class InMemoryAccountRepository implements AccountRepository {
    private storage = new Map<string, Account>();

    find(id: string): Account | undefined {
        return this.storage.get(id);
    }

    save(account: Account): void {
        this.storage.set(account.id, account);
    }
}
